crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1031,6 +1031,56 @@ impl Analyzer<'_, '_> {
                 return Ok(());
             }
 
+            (
+                Type::Tuple(Tuple {
+                    span: tspan,
+                    elems,
+                    metadata,
+                    tracker,
+                }),
+                Type::Interface(Interface {
+                    span: ispan,
+                    name,
+                    type_params,
+                    extends,
+                    body,
+                    metadata: imetadata,
+                    tracker: itracker,
+                }),
+            ) => {
+                let lhs = elems.iter().map(|TupleElement { span, label, ty, tracker }| ty);
+                for parent in extends {
+                    let params = parent.type_args.clone().unwrap().params;
+                    let rhs = self
+                        .type_of_ts_entity_name(span, &parent.expr, parent.type_args.as_deref())?
+                        .freezed();
+                    if let Some(Array {
+                        span,
+                        elem_type,
+                        metadata,
+                        tracker,
+                    }) = rhs.array()
+                    {
+                        if let Some(Union {
+                            span,
+                            types,
+                            metadata,
+                            tracker,
+                        }) = elem_type.union_type()
+                        {
+                            for (lhs, ty) in lhs.clone().zip(types) {
+                                if let (Some(KeywordType { kind: lkind, .. }), Some(KeywordType { kind: rkind, .. })) =
+                                    (lhs.to_owned().keyword(), ty.keyword())
+                                {
+                                    println!("@@@\n{lkind:#?}:{rkind:#?}");
+                                    let diff = lkind == rkind;
+                                    println!("@@@\n{diff}");
```

Comment:
1. Is there any existing function to compare two KeywordType and return reasonable error?
2. Do you think I'm going correct way?

Reply:
1. You can use `assign_with_opts(...)?` to do it
2. I think you should not reimplement whole logic here, but delegate to a recursive call to `assign_with_opts`

Reply:
It looks like we should keep logic here.
Otherwise, it breaks other tests
https://github.com/dudykr/stc/actions/runs/4110781244/jobs/7093929846